idd/Energy+.idd.in

```diff
@@ -73923,6 +73923,321 @@ HeatPump:PlantLoop:EIR:Heating,
+       \key NotModulated
+       \key ConstantFlow
+       \key LeavingSetpointModulated
+       \default NotModulated
+       \note Flow Mode for the water side of the fuel-fired absorption heat pump
+  A9 , \field Outdoor Air Temperature Curve Input Variable
+       \required-field
+       \type choice
+       \key DryBulb
+       \key WetBulb
+       \default DryBulb
+       \note Outdoor air temperature curve input variable;
+       \note The options are Outdoor Air Dry Bulb or Wet Bulb temperature for curves
+  A10, \field Water Temperature Curve Input Variable
+       \required-field
+       \type choice
+       \key EnteringCondenser
+       \key LeavingCondenser
+       \default EnteringCondenser
+       \note Water Temperature curve input variable - Condenser Entering or Leaving Water Temperature for curves
+  A11, \field Normalized Capacity Function of Temperature Curve Name
+       \required-field
+       \type object-list
+       \object-list BivariateFunctions
+       \note: CAPFT - Normalized Capacity Function of Temperature Curve name,
+       \note which is a biquadratic curve or a lookup table.
+  A12, \field Fuel Energy Input Ratio Function of Temperature Curve name
+       \required-field
+       \type object-list
+       \object-list BivariateFunctions
+       \note EIRFT - Fuel Energy Input Ratio Function of Temperature Curve name,
+       \note which is a biquadratic curve or a lookup table.
+  A13, \field Fuel Energy Input Ratio Function of PLR Curve Name
+       \required-field
+       \type object-list
+       \object-list UnivariateFunctions
+       \note EIRFPLR - Fuel Energy Input Ratio Function of Part Load Ratio(PLR) Curve name,
+       \note which is a cubic curve or a lookup table.
+  N6 , \field Minimum Part Load Ratio
+       \minimum> 0.0
+       \maximum 1.0
+       \default 0.1
+       \note Minimum Part Load Ratio (PLR) in between 0 and 1
+  N7 , \field Maximum Part Load Ratio
+       \minimum> 0.0
+       \maximum 1.0
+       \default 1.0
+       \note Maximum Part Load Ratio (PLR) in between 0 and 1
+  A14, \field Fuel Energy Input Ratio Defrost Adjustment Curve name
+       \type object-list
+       \object-list UnivariateFunctions
+       \note EIRDEFROST - Energy Input Ratio Defrost Adjustment Curve name,
+       \note which is a cubic curve or a lookup table function of Outdoor Air Temperature.
+  A15, \field Defrost Control Type
+       \type choice
+       \key timed
+       \key OnDemand
+       \default timed
+       \note Defrost operation control type: timed or OnDemand
+  N8 , \field Defrost Operation Time Fraction
+       \minimum 0
+       \maximum 1
+       \default 0
+       \note Defrost operation time fraction, which will be used for timed defrost control type.
+  N9 , \field Resistive Defrost Heater Capacity
+       \units W
+       \minimum 0
+       \default 0
+       \note The heating power of electric resistive defrost heater. Only used for timed defrost control type.
```

Comment:
If this is only used for time defrost, does that mean "OnDemand" defrost is a reverse cycle? Is "Fuel Energy Input Ratio Defrost Adjustment Curve name" used only for OnDemand?

Reply:
My understanding is that since it is a fuel-fired machine, it would probably easier to do some kind of "gas by pass" to defrost--this also looks more straight forward since it is like just use some part of the fuel energy to defrost directly (equivalent to a small defrost burner). I tend to believe technically to do "reverse cycle" might be much more complex to do here?

But anyway, here we did put a restriction that when "OnDemand" is used, then the "Fuel Energy Input Ratio Defrost Adjustment Curve Name" method is used. If "Timed" is used, an electric resister (although sounds weird for a fuel-fired machine) is used and then the defrost time fractions could be active.

Reply:
Then I would move Fuel Energy Input Ratio Defrost Adjustment Curve Name to be after the Resistive Defrost Heater Capacity field and edit the notes to indicate this is only used with OnDemand.

I would also consider changing "Timed" to "TimedResistive" to help clarify. I don't feel strongly on this one.